frontend/static_vol/js/components/Dashboard.js

```diff
@@ -75,13 +93,118 @@ export default class Dashboard extends PageBase {
+    }
+
+    updateAvatar() {
+        const inputFile = document.getElementById('inputAvatarFile');
+        inputFile.click();
+    }
+
+    listenPickFileAvatar() {
+        const inputFile = document.getElementById('inputAvatarFile');
+        const boundPickFileAvatar = this.pickFileAvatar.bind(this);
+        this.addListListenInInstance(inputFile, boundPickFileAvatar, 'change');
+    }
+    
+    pickFileAvatar(ev) {
+        const imgAvatar = document.getElementById('imgAvatar');
+        const btnWrapUpdateAvatar = document.querySelector('.blockAvatar_button');
+        const listButton = document.querySelector('.blockAvatar_listButton');
+        const file = ev.target.files[0];
+        if (file) {
+            const fileReader = new FileReader();
+            fileReader.onload = (ev) => {
+                imgAvatar.src = ev.target.result;
+                btnWrapUpdateAvatar.classList.remove('is-show');
+                listButton.classList.add('is-show');
+                this.listenUploadAvatar();
+            };
+            fileReader.readAsDataURL(file);
+        }
+    }
+
+    listenUploadAvatar() {
+        const btnAvatarCancel = document.getElementById('btnAvatarCancel');
+        const btnAvatarUpload = document.getElementById('btnAvatarUpload');
+        const boundCancelAvatar = this.cancelAvatar.bind(this);
+        const boundUploadAvatar = this.uploadAvatar.bind(this);
+
+        addListenerToList(
+            this.listListenUploadAvatar, btnAvatarCancel, boundCancelAvatar, 'click');
+        addListenerToList(
+            this.listListenUploadAvatar, btnAvatarUpload, boundUploadAvatar, 'click');
+    }
+
+    async uploadAvatar(ev) {
+        ev.preventDefault();
+        const inputFile = document.getElementById('inputAvatarFile');
+        const formData = new FormData();
+        const file = inputFile.files[0];
+        if (!file) {
+            return;
+        }
+        formData.append('avatar', file);
+
+        const accessToken = getToken('accessToken');
+        if (accessToken === null) {
+            return Promise.resolve(null);
+        }
+        fetch('/api/players/avatar/', {
+                method: 'PUT',
+                headers: {
+                    'Authorization': `Bearer ${accessToken}`,
+                },
+                body: formData
+        })
+            .then( async (response) => {
+                if (!response.ok) {
+                    const responseBody = await response.text();
+                    throw new Error(responseBody);
+                 }
+                return response.json();
+            })
+            .then( async (data) => {
+                this.siteInfo.setAvatar(data.newAvatar);
+                this.cancelAvatar();
+                await switchDisplayAccount();
+                addNotice('アバターを変更しました', false);
+            })
+            .catch((error) => {
+                this.cancelAvatar();
+                console.error('Error:', error);
```

Comment:
```suggestion
                console.error('Error:', error);
                addNotice('エラーが発生しました', true);
```
エラーがあったことを通知すると分かりやすそうです